src/lightning/data/streaming/dataset.py

```diff
@@ -194,9 +208,13 @@ def __next__(self) -> Any:
 
 
 def _try_create_cache_dir(input_dir: str, shard_rank: int = 0) -> Optional[str]:
-    if "LIGHTNING_CLUSTER_ID" not in os.environ or "LIGHTNING_CLOUD_PROJECT_ID" not in os.environ:
-        return None
     hash_object = hashlib.md5(input_dir.encode())
+    if "LIGHTNING_CLUSTER_ID" not in os.environ or "LIGHTNING_CLOUD_PROJECT_ID" not in os.environ:
+        cache_dir = os.path.join(
+            os.path.expanduser("~"), ".lightning", "chunks", hash_object.hexdigest(), str(shard_rank)
```

Comment:
we could define the chunks base folder in constants.py

and using `Path.home()` would be nice.

Reply:
done !